lib/color/color.go

```diff
@@ -14,3 +77,66 @@ func Darken(colorString string) (string, error) {
 	// decrease luminance by 10%
 	return colorful.Hsl(h, s, l-.1).Clamped().Hex(), nil
 }
+
+func LuminanceCategory(colorString string) (string, error) {
+	l, err := Luminance(colorString)
+	if err != nil {
+		return "", err
+	}
+
+	switch {
+	case l >= .88:
+		return "bright", nil
+	case l >= .55:
+		return "normal", nil
+	case l >= .30:
+		return "dark", nil
+	default:
+		return "darker", nil
+	}
+}
+
+func Luminance(colorString string) (float64, error) {
+	c, err := csscolorparser.Parse(colorString)
+	if err != nil {
+		return 0, err
+	}
+
+	l := float64(
+		float64(0.299)*float64(c.R) +
```

Comment:
where'd this come from? can you leave the source as a comment?

Reply:
alright